test/acceptance/test/utils.go

```diff
@@ -845,3 +846,18 @@ func getGitProvider(org string, repo string, providerName gitproviders.GitProvid
 
 	return gitProvider, orgRef, err
 }
+
+func setGitProvider() (gitproviders.GitProviderName, string, string) {
+	gitlab := "gitlab"
+	github := "github"
+	gitProvider := os.Getenv("GIT_PROVIDER")
```

Comment:
But the tests won't fail even if the env is not set. The function returns `github` by default even if the env is missing. Si the tests would run on github by default. The only change this function brings is if we want to run all the github tests in gitlab repos. I don't want this variable set as a requirement because I avoided using this env for specific gitlab tests.

Reply:
That makes sense. I would argue then that we should have some doc explaining this behavior and also how to run all this cases, like:
- How to run github tests in gitlab
- How to run gitlab tests
- How to run github tests

BTW if you need to have a different value for specific variable in one of the tests you can add it to that specific test only. Like here for instance https://github.com/weaveworks/weave-gitops/blob/main/test/integration/server/remove_test.go#L254. This `AutoMerge` variable is not needed in all cases only on that specific one. If you ever need to use something specific in more than one test, and you want to avoid writting it twice , you could add a  new Context and set that variable's value in a BeforeEach within that context. Like this for instance https://github.com/weaveworks/weave-gitops/blob/main/test/integration/server/remove_test.go#L223. `gitlabGroup ` was written once but it runs for each test within that context.

Reply:
I can definitely update the ReadME file for clarification

Reply:
I'll update the ReadME in a follow up PR and try to see if setting git-provider variables at `BeforeSuite` works. For now, I'm merging this PR.

Reply:
NVM, I updated the readme and set the variables at suite level.